src/components/input/select/index.tsx

```diff
@@ -308,7 +309,7 @@ export class SmoothlyInputSelect implements Input, Editable, Clearable, Componen
 		return (
 			<Host
 				tabIndex={0}
-				class={{ "has-value": this.selected.length !== 0, open: this.open }}
+				class={{ "has-value": this.selected.length !== 0, open: this.open, invalid: !!this.invalid }}
```

Comment:
shouldn't we just reflect invalid? maybe i forget how it works

Reply:
I can.